src/pepr/operator/crd/validators/package-validator.ts

```diff
@@ -141,5 +141,24 @@ export async function validator(req: PeprValidateRequest<UDSPackage>) {
     }
   }
 
+  const monitors = pkg.spec?.monitor ?? [];
+
+  // Ensure serviceMonitors use a unique description or selector/portName used for generating the resource name
+  const monitorNames = new Set<string>();
+
+  for (const monitor of monitors) {
+    const monitorName = sanitizeResourceName(
+      monitor.description || `${Object.values(monitor.selector)}-${monitor.portName}`,
+    );
```

Comment:
Realized in retrospect we could just import [`generateMonitorName`](https://github.com/defenseunicorns/uds-core/blob/monitor-validate/src/pepr/operator/controllers/monitoring/common.ts#L4) and use that here to ensure we keep this logic in sync with our name generation rather than duplicating.